src/plugins/plugin.legend.js

```diff
@@ -78,17 +78,21 @@ defaults._set('global', {
 	},
 
 	legendCallback: function(chart) {
-		var text = [];
-		text.push('<ul class="' + chart.id + '-legend">');
-		for (var i = 0; i < chart.data.datasets.length; i++) {
-			text.push('<li><span style="background-color:' + chart.data.datasets[i].backgroundColor + '"></span>');
+		var list = document.createElement('ul');
+		var i, ilen, listItem, listItemSpan;
```

Comment:
While at it, I'd cache `var datasets = chart.data.datasets`;

Reply:
done